src/components/NanoContract/SelectAddressModal.component.js

```diff
@@ -57,7 +57,9 @@ export const SelectAddressModal = ({ address, show, onSelectAddress, onDismiss }
   };
 
   const onSelectItem = (item) => {
-    setSelectedItem(item);
+    if (item !== address) {
+      setSelectedItem(item);
+    }
     toggleEditAddressModal();
```

Comment:
I'm not sure if I understand correct but you are closing the modal if the user clicks on the already selected address? If that's the case, I prefer you don't do like this, we should keep the modal opened.

Reply:
Actually, this is only called by AddressItem when current address is different from the item address. We can safely remove this condition. My bad.